### PR TITLE
gitignore updates

### PR DESCRIPTION
**Pull request by nCastle1**
These changes to the gitignore file should help prevent Resources.designer.cs, Thumbs.db, .DS_Store files, and others from being included in git diffs. It also removes existing resource.designer.cs files. Finally, it brings our gitignore in line with the standard Visual Studio gitignore while retaining some existing customizations.

### CODE REVIEW

**Comment by nCastle1** [comment]
@mbranscomb per our discussions I've updated the gitignore file. I primarily based it on the latest version of the standard Visual Studio gitignore, but I added some things specific to Xamarin & macOS development from around the web. I also included some things for common tools that I saw as recurring themes in my research (Rider, JustCode, TeamCity, DotCover, MightyMoose, IntelliJ). I also included some for tools that developers are likely to use but that tend to leave behind artifacts (Python, Node.js).

File: .gitignore [hunk 1]
```diff
@@ -76,43 +105,102 @@ DocProject/Help/html
...
 ~$*
 *.dbmdl
+node_modules/
+typings/
+.fake/
+
 Generated_Code #added for RIA/Silverlight projects
 
 # Backup & report files from converting an old project file to a newer
 # Visual Studio version. Backup files are not needed, because we have git ;-)
```
Comment (mbranscomb) [inline .gitignore]: Unnecessary comment? `Backup files are not needed, because we have git ;-)`

File: .gitignore [hunk 2]
```diff
@@ -76,43 +105,102 @@ DocProject/Help/html
...
 
-#Documentation file
+# Python
+*.pyc
+__pycache__/
+
+# Documentation file
 SampleDescriptions.xml
 /src/.vs/config/applicationhost.config
 /src/Windows/ArcGISRuntime.Windows.Samples/project.lock.json
```
Comment (mbranscomb) [inline .gitignore]: Do we still need these specific excludes?

```
/src/Windows/ArcGISRuntime.Windows.Samples/project.lock.json
/src/Windows/ArcGISRuntime.Windows.Viewer/project.lock.json
src/Windows/ArcGISRuntime.Windows.Samples/Samples/Security/IntegratedWindowsAuthVB/.vs/
src/Windows/ArcGISRuntime.Windows.Samples/Samples/Security/OAuth/project.lock.json
src/Windows/ArcGISRuntime.Windows.Samples/Samples/Security/TokenSecuredKnownUserVB/project.lock.json
src/Windows/ArcGISRuntime.Windows.Samples/Samples/Security/TokenSecuredChallengeVB/.vs/
```

File: .gitignore [hunk 3]
```diff
@@ -7,16 +7,22 @@ TestResults
...
 *.user
 *.sln.docstates
 .vs/
 
+
 # Build results
 [Dd]ebug/
 [Rr]elease/
+bld/
 x64/
```
Comment (mbranscomb) [inline .gitignore]: x86 too?

File: .gitignore [hunk 4]
```diff
@@ -76,43 +105,102 @@ DocProject/Help/html
 # Click-Once directory
 publish
```
Comment (mbranscomb) [inline .gitignore]: Should this be `publish/` ?

File: .gitignore (hunk 1 above)
Comment (nCastle1) [inline .gitignore]: That was in the template gitignore, but I'll get rid of it.

**Pull request merged**